t/fonts/mathaccents.tex

```diff
@@ -16,4 +16,35 @@
 $\widetilde{aaa}$   & $\mathaccent"0365{aaa}$ \\
 $\widehat{aaa}$     & $\mathaccent"0362{aaa}$ \\
 \end{tabular}
+
+\def\abc{ABC}
+\def\nothing{}
+Test what arguments hat accepts
+(should be same with check, breve, acute, grave, tilde, bar, vec,
+dot, ddot, overline, widehat, widetilde)
+\def\testmacro{\hat}
+\[ [\testmacro A] \quad
+ [\testmacro{}] \quad
+ [\testmacro{A}] \quad
+ [\testmacro{ABC}] \quad
+ [\testmacro\nothing A] \quad
+ [\testmacro\relax A] \quad
+ [\testmacro{\verb+$+}] \quad
+ [\testmacro\bgroup A \egroup] \quad
+ [\testmacro\bgroup \verb+$+ \egroup] \]
+Note that this only accents the A
+\[ [\testmacro\abc] \]
+
+Other accent-like macros (underbrace, overbrace) only accept regular style arguments
```

Comment:
kind of a over-protective question but, maybe nice to also have them tested stacked? namely the two:
```tex
\[ \overbrace{\underbrace{ARG}_{under}}^{over} \]
\[ \underbrace{\overbrace{ARG}^{over}}_{under} \]

```

Reply:
I could add that, but it's kinda testing something entirely different. The added tests question what kind of material various cs (in LaTeX) can process as "arguments", and make sure that LaTeXML processes them in the same way.  Since over/underbrace actually turn out to only accept the standard  `#1` style argument (ie a single token or {} delimited), and your sample has the args in braces, it "has to" work.  Of course, something else could go wrong :>

Reply:
Alright, maybe its point is indeed too different for this PR. I tried a few other tests, one of which had a discrepancy with pdflatex, but again in an unrelated way:
```tex
\testmacro\noexpand\relax
```
which produces:
```
Error:expected:Token Missing argument Token for Core::Definition::Expandable[\noexpand Token] at ubrace.tex; line 4 col 24
```

This ran just fine in pdflatex and produced an underbrace over some empty space - for which I wonder how the width was determined. Leaving it to you what to do with this partial information :>

Reply:
An extension to my previous note, if you do:
```tex
\[ [\expandafter\underbrace\underbrace\noexpand\relax] \]
```

pdflatex will produce two adjacent empty undebraces fenced in with the square brackets, without any errors. latexml instead emits a warning for a missing argument, and produces HTML with a `missing` word with two vertical underbraces under it. So this may be more directly related to the `{}` argument, even if my test is admittedly tortured.

Reply:
The last patch deals with the basic `\noexpand\relax`, but I'm not even sure that the last cases are even fixable, given that LaTeXML tries to get "an argument" and process it, whereas TeX just kinda grinds on through as long as it can. That it puts an accent on nothing indicates that it's not a realistically working case; it's kinda surprising that it doesn't create an error.  Anyway: I'm punting (for now).

Reply:
Thanks for the `\noexpand` bit. For the `\expandafter` bit I went to check the source, and indeed there is a difference - macros such as `\int`, `\eq`, `\neq`, `\mapsto`, `\cong`, `\longrightarrow`, `\cdots`, `\overbrace`, `\underbrace` ... in `latex/base/fontmath.ltx` are declared as protected, while our DefMath declarations do not have the protection added (for now?)

I think every argument-accepting math macro in `fontmath.ltx` ends up declared as protected.

Indeed, just quickly changing the definition to:
```perl
DefMath('\lx@math@underbrace {}', "\x{23DF}", operator_role => 'UNDERACCENT',     # BOTTOM CURLY BRACKET
  scriptpos => 'mid');
DefMacroI('\underbrace', undef, Tokens(T_CS('\protect'), T_CS('\lx@math@underbrace')));
```

leads to my last example correctly finding the letter `f` as the argument of underbrace, as well as the rest of the details that come up through pdflatex. So that may be the only fix needed to get all my examples working - though what the "Best" way to implement the protection is, I am unsure. Maybe we need a `protected => 1` flag for DefMath, which is turned on by default?

Reply:
Also, feel free to merge the PR as-is, I can open an issue for this newly discovered discrepancy.